src/main/java/org/jabref/model/database/BibDatabase.java

```diff
@@ -293,16 +293,22 @@ public synchronized void addString(BibtexString string) throws KeyCollisionExcep
     }
 
     public void addStrings(Collection<BibtexString> stringsToAdd) {
+
```

Comment:
I would replace this method by a `setStrings` method, that simply replaces the internal list of strings by the given one (maybe after validating that there are no duplicate keys). In addition, there should be a validation directly in the dialog.

Reply:
I implemented a validation int he dialog and replaced this by setSTrings